File: packages/android/connector/src/main/java/app/quiltt/connector/TelemetryUtils.kt
package app.quiltt.connector

import android.os.Build

object TelemetryUtils {
    fun getSDKAgent(sdkVersion: String, platformInfo: String): String {
        return "Quiltt/$sdkVersion ($platformInfo)"
    }

    fun getRuntimePlatformInfo(): String {
        val release = Build.VERSION.RELEASE ?: "Unknown"
        val apiLevel = Build.VERSION.SDK_INT
        val manufacturer = Build.MANUFACTURER ?: "Unknown"
        val model = Build.MODEL ?: "Unknown"

        return "Android; OS $release (API $apiLevel); Device $manufacturer $model"
    }
}

File: packages/ios/Sources/QuilttConnector/TelemetryUtils.swift
import Foundation

#if canImport(UIKit)
    import UIKit
#endif

enum TelemetryUtils {
    static func getSDKAgent(sdkVersion: String, platformInfo: String) -> String {
        return "Quiltt/\(sdkVersion) (\(platformInfo))"
    }

    static func getRuntimePlatformInfo() -> String {
        #if canImport(UIKit)
            let device = UIDevice.current
            let osName = device.systemName
            let osVersion = device.systemVersion
            let model = device.model
            return "iOS; OS \(osName) \(osVersion); Device \(model)"
        #else
            let osVersion = ProcessInfo.processInfo.operatingSystemVersionString
            return "Apple; OS \(osVersion)"
        #endif
    }
}
